docs/content/developers/design-keuzes.md

```diff
@@ -190,3 +190,17 @@ RGBZ. BESLUIT krijgt een optionele relatie naar ZAAK.
 
 De verwachting is dat er later vergelijkbare designkeuzes gemaakt worden voor
 andere resources, zoals bijvoorbeeld klantcontacten.
+
+## Zaak afsluiten
+
+In de huidige ZDS 1.x standaard is er nog [geen eenduidig besluit genomen over hoe een zaak wordt afgesloten](https://discussie.kinggemeenten.nl/discussie/gemma/koppelvlak-zs-dms/afsluiten-van-een-zaak). 
+
+Wij hebben gekozen voor grofweg de volgende aanpak:
+* Objecttype STATUSTYPE krijgt een nieuw attribuut: `isEindStatus` (ja/nee). Er mag maximaal 1 eindstatus zijn per ZAAKTYPE.
+* Het attribuut `Einddatum` op ZAAK wordt alleen-lezen.
+* Bij het toekennen van een STATUSTYPE met `isEindStatus=ja` aan ZAAK wordt de `ZAAK.Einddatum` automatisch ingevuld met de waarde uit `STATUS.datumStatusGezet`.
+
+Het RGBZ wordt hier op aangepast.
+
+**TODO:**
+_Of de ZAAK hierna niet meer kan worden aangepast is wellicht praktisch niet handig maar zou wel het zuiverste zijn. Het is alleen technisch niet 100% af te dekken. Je kan namelijk een document toevoegen aan een zaak, zonder dat het ZRC "dit weet" aangezien de relatie (op dit moment) ligt bij het DRC._
```

Comment:
Ik ga hier dus eigenlijk mee met @ArjanKloosterboer - interactie met een zaak moet valideren dat de zaak (nog) niet afgesloten is.

Concreet betekent dit:

1. Ik kan een nieuw informatieobject toevoegen via de API. Hier vindt geen validatie plaats op de zaak, want een informatieobject kan prima bestaan zonder zaak.
2. Ik probeer dit informatieobject te relateren aan een afgesloten zaak. Het DRC (want - daar wordt de relatie vastgelegd) vraagt de zaak in kwestie op van het ZRC en gaat na dat deze niet afgesloten is. Het DRC stelt vast dat de zaak wel afgesloten is, en geeft een HTTP 400 error, met in de foutbody de melding dat de zaak al afgesloten is, en dat er daarom geen nieuwe informatieobjecten kunnen gerelateerd worden.

Het volgende moet wel mogelijk blijven:

1. Uitlezen van de relatie
2. Bijwerken van het informatieobject (PUT/PATCH). Hoe dit er concreet moet gaan uitzien, moet uitgewerkt worden met het versioneren/tijdreizen.

Worden geblokkeerd zodra de zaak gesloten is:

1. Verwijderen van zaakinformatieobjectrelatie. Dit wordt een probleem als er na het afsluiten van de zaak correcties moeten gebeuren omdat er een fout informatieobject aan de zaak gerelateerd was.
2. Bijwerken van de relatie.
3. Toevoegen van nieuwe relaties.

Reply:
In overleg kwam hier uit dat de API niet zomaar een zaak read-only moet maken. Het moet namelijk mogelijk blijven een Zaak te heropenen en/of toch aan te passen, want .... Dit wordt daardoor meer en autorisatie verhaal. Ik kies er daarom voor om dit niet binnen scope op te pakken. Zaken blijven gewoon bewerkbaar op API niveau voor nu.

Rekening houden met tijdreizen heb ik nog niet in User Stories gezien dus laat ik voor nu buiten beschouwing.